### PR TITLE
Properly format log messages

### PR DESCRIPTION
**Pull request by cjellick**
addresses https://github.com/rancher/rancher/issues/1385

I'll try to explain the fix:
-  When Vince get logs from host-api, we prepend every websocket message we send to him with either 00 01 or 02 to indicate if the message is coming from stdout, stderr, or a combined stream.
- Vince assumes that such a prefix is at the beginning of every websocket message.
- The bug was that I would send a websocket message every time I encountered a newline in the docker logs but was only prepending `00` each time docker sent me a message chunk and the two did not always match up. So, I would sometimes send messages that didn't have the prefix.

You could reproduce the bug by doing this:
```
docker run -it ubuntu bash
echo -e "Foo\n bar"
```
and then viewing the logs for that container in the rancher UI.
When I hit the `\n`, I sent a new websocket message but `bar` was not prepended with `00`.

So, to fix I wrote a custom split function for my scanner instead of using the default ScanLines one that split on new lines. Now, when i prepend 00 to a message, I also add the suffix `[RANLOGS]` and my custom split function splits on that token.

### CODE REVIEW

**Comment by cjellick** [comment]
You'll notice three commits. The third `Fix unreliable stats test` is unrelated to my change but it fixes a test that would fail inconsistently because it was relying on non-existing filter-by-image functionality in the docker api.

**Comment by cjellick** [comment]
I also added the go-check testing library for more sane tests.

File: logs/logWrapper.go [hunk 1]
```diff
@@ -16,16 +20,22 @@ type stdbothWriter struct {
 }
 
 func (w stdoutWriter) Write(message []byte) (n int, err error) {
-	n, err = w.writer.Write(append([]byte("01 "), message...))
-	return n - 3, err
+	msg := append([]byte("01 "), message...)
```
Comment (alena1108) [inline logs/logWrapper.go]: Nit: looks like we can extract repeatable code (23-26, 30-33, 37-40) to its own method?

File: logs/logs_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		if err != nil {
+			if err == io.EOF {
+				return
+			}
+			c.Fatal(err)
+		}
+		msgStr := string(msg)
+		log.Infof("MESSAGE %v: |%s|", count, msgStr)
+		if !strings.HasPrefix(msgStr, "00") {
+			//		c.Fatalf("Message didn't have 00 prefix: [%s]", msgStr)
```
Comment (alena1108) [inline logs/logs_test.go]: Do we expect to reenable this commented code?

Comment (cjellick) [inline logs/logs_test.go]: whoops. thanks

**Comment by alena1108** [comment]
LGTM

**Comment by ibuildthecloud** [comment]
LGTM

**Pull request merged**